test/IntegrationTests/TicketService/test_delete_custom_field.py

```diff
@@ -0,0 +1,219 @@
+
+class TestDeleteCustomFieldAPI(unittest.TestCase):
+    BASE_URL = "http://localhost:8000"
+
+    allowed_custom_field_sql_types = ["VARCHAR(255)", "INT", "BOOLEAN", "DATE", "MEDIUMTEXT", "FLOAT", "TEXT"]
+
+    def setUp(self):
+        """Setup function to initialize valid customer GUID and custom fields."""
+        logger.info("=== Initializing test setup ===")
+
+        # Step 1: Create a new customer
+        customer_url = f"{self.BASE_URL}/addcustomer"
+        response = requests.post(customer_url)
+        self.assertEqual(response.status_code, HTTPStatus.OK, "Failed to create customer")
+        self.valid_customer_guid = response.json().get("customer_guid")
+        logger.info(f"Customer created with GUID: {self.valid_customer_guid}")
+
+        logger.info("Setup complete.")
+        logger.info(f"Starting test: {self._testMethodName}")
+
+    def list_custom_fields(self):
+        """Reusable function to list all custom fields."""
+        logger.info("Listing all custom fields.")
+        url = f"{self.BASE_URL}/custom_fields?customer_guid={self.valid_customer_guid}"
+        response = requests.get(url)
+
+        if response.status_code == HTTPStatus.NOT_FOUND:
+            logger.info(f"No custom fields found for customer {self.valid_customer_guid}")
+            return []  # Return an empty list if no custom fields are found
+        else:
+            self.assertEqual(response.status_code, HTTPStatus.OK, "Failed to list custom fields")
+            return response.json()
+
+    def test_add_and_delete_all_custom_fields(self):
+        """Test adding all allowed custom fields, deleting them one by one, and verifying the results."""
+        custom_field_url = f"{self.BASE_URL}/custom_fields"
+
+        # Step 1: Add custom fields dynamically
+        custom_fields = []
+        for i, field_type in enumerate(self.allowed_custom_field_sql_types):
+            if field_type.startswith("VARCHAR"):
+                field_name = f"varchar_{field_type[7:].replace('(', '').replace(')', '')}_field"
+            else:
+                field_name = f"{field_type.lower()}_field"
+
+            custom_fields.append({
+                "field_name": field_name,
+                "field_type": field_type,
+                "required": i % 2 == 0
+            })
+
+        for field in custom_fields:
+            logger.info(f"Adding custom field: {field['field_name']}")
+            field_data = {**field, "customer_guid": self.valid_customer_guid}
+            response = requests.post(custom_field_url, json=field_data)
+            self.assertEqual(
+                response.status_code,
+                HTTPStatus.CREATED,
+                f"Failed to add custom field {field['field_name']}"
+            )
+
+        # Step 2: Verify all custom fields are added
+        initial_fields = self.list_custom_fields()
+        self.assertEqual(len(initial_fields), len(custom_fields), "Custom fields count mismatch after addition")
```

Comment:
Can we also check the names of the custom fields and its field type here and compare it with the input?

Reply:
modified sir